fpsim/locations/nuhdss.py

```diff
@@ -0,0 +1,1252 @@
+        [15.0, 0.14377365580688461],
+        [16.0, 0.13271744734209995],
+        [17.0, 0.11915611658325072],
+        [18.0, 0.13735481818469894],
+        [19.0, 0.0841039265081519],
+        [20.0, 0.07725867074164659],
+        [21.0, 0.03982337306065369],
+        [22.0, 0.031195559243867545],
+        [23.0, 0.020750304422300126],
+        [24.0, 0.014468030815585422],
+        [25.0, 0.010870195645684769],
+        [26.0, 0.007574195696769944],
+        [27.0, 0.0034378402773621282],
+        [28.0, 0.0031344552061394622],
+        [29.0, 0.0018168079578966389],
+        [30.0, 0.001385356426809007],
+        [31.0, 0.0004912818135032509],
+        [32.0, 0.00045904179812542576],
+        [33.0, 0.0005049625590548578],
+        [34.0, 0.000165858204720886],
+        [35.0, 0.00019259487032758347],
+        [36.0, 0.0002126920535675137],
+        [37.0, 8.84428869703282e-05],
+        [38.0, 5.07209448615522e-05],
+        [39.0, 6.555458199225806e-05],
+        [41.0, 0.00013980442816424654],
+        [44.0, 4.372731039149624e-05]])
+        '''
+    sexual_debut = pd.read_csv(thisdir / 'nuhdss' / 'sex_debut.csv')
+    debut_age = {}
+    debut_age['ages'] = sexual_debut.Age_at_first_sex.values
+    debut_age['probs'] = sexual_debut.prob.values
+
+    return debut_age
+
+
+def exposure_age():
+    '''
+    #Ask
+    Returns an array of experimental factors to be applied to account for
+    residual exposure to either pregnancy or live birth by age.  Exposure to pregnancy will
+    increase factor number and residual likelihood of avoiding live birth (mostly abortion,
+    also miscarriage), will decrease factor number
+    '''
+    exposure_correction_age = np.array([[0, 5, 10, 12.5, 15, 18, 20, 25, 30, 35, 40, 45, 50],
+                                        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]])
+    exposure_age_interp = data2interp(exposure_correction_age, fpd.spline_preg_ages)
+
+    return exposure_age_interp
+
+
+def exposure_parity(): #Consult
+    '''
+    Returns an array of experimental factors to be applied to account for residual exposure to either pregnancy
+    or live birth by parity.
+    '''
+    exposure_correction_parity = np.array([[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 20],
+                                           [1, 1, 1, 1, 1, 1, 1, 0.8, 0.5, 0.3, 0.15, 0.10, 0.05, 0.01]])
+    exposure_parity_interp = data2interp(exposure_correction_parity, fpd.spline_parities)
+
+    return exposure_parity_interp
+
+
+def birth_spacing_pref():
+    '''
+    Returns an array of birth spacing preferences by closest postpartum month.
+    Applied to postpartum pregnancy likelihoods.
+
+    NOTE: spacing bins must be uniform!
+    '''
+    postpartum_spacing = np.array([
+        [0, 1],
+        [3, 1],
+        [6, 1],
+        [9, 1],
+        [12, 1],
+        [15, 1],
+        [18, 1],
+        [21, 1],
+        [24, 1],
+        [27, 1],
+        [30, 1],
+        [33, 1],
+        [36, 1],
+    ])
+
```

Comment:
By the way, this postpartum spacing can also be used in your calibration to adjust mismatch in birth spacing specifically. It's a multiplier so when set to 1, it has no effect. Most of the time we underestimate the very long (36+) birth intervals.